workerpool/pool.go

```diff
@@ -206,6 +206,9 @@ func (w *worker) run() {
 				w.log.Errorw("closing source failed", "workerID", w.id, "error", err.Error())
 			}
 			return
+		case <-time.After(1 * time.Second):
```

Comment:
Should this be a configurable at some point?

Reply:
Why would anyone want to configure that?